packages/idempotency/src/persistence/PersistenceLayer.ts

```diff
@@ -10,22 +11,30 @@ abstract class PersistenceLayer implements PersistenceLayerInterface {
   private envVarsService!: EnvironmentVariablesService;
 
   private expiresAfterSeconds: number;
-
-  private functionName: string = '';
-
+  
   private hashDigest: BinaryToTextEncoding;
-
+  
   private hashFunction: string;
+  
+  private idempotencyKeyPrefix: string;
 
   public constructor() { 
     this.setEnvVarsService();
     this.expiresAfterSeconds = 60 * 60; //one hour is the default expiration
     this.hashFunction = 'md5';
     this.hashDigest = 'base64';
-        
+    this.idempotencyKeyPrefix = this.getEnvVarsService().getFunctionName();
+
   }
-  public configure(functionName: string = ''): void {
-    this.functionName = this.getEnvVarsService().getLambdaFunctionName() + '.' + functionName;
+
+  /**
+   * Configures the persistence layer by passing the name of the idempotent function. This will be used
+   * in the prefix of the idempotency key
+   * 
+   * @param {PersistenceLayerConfigureOptions} options - configuration object for the persistence layer
+   */
+  public configure(options?: PersistenceLayerConfigureOptions): void {
+    if (options?.functionName && options.functionName.trim() !== '') this.idempotencyKeyPrefix = `${this.idempotencyKeyPrefix}.${options.functionName}`;
```

Comment:
(Question) 
Now that I review this function again. It seems like it uses both two lambda names (one from environment variable  + another from options object).   

I looked at Python (see the image below), it is doing the same thing.  Do you know the root cause? (If not, don't worry about it, I'll check with Python team myself)

<img width="898" alt="image" src="https://user-images.githubusercontent.com/61999/201926143-4efed04b-8991-4fd6-8230-d8b7943dd2c6.png">

Reply:
Yes, I was very confused by this as well and during my first refactor (not pushed) I actually removed this, but then reintroduced it.

From the way I understand it, the first one (that comes from the environment in our implementation) is the name of the Lambda resource as it appears in AWS Lambda.
The second one instead is a name that can be set and refers to the function being made idempotent in your code.

Imagine that you have a function called `MyLambda` like this:
```ts
const processPayment = async () => {
  // ... do stuff
};

export const handler = async () => {
  const result = makeIdempotentFn(await processPayment());
};
```

In this case, the idempotency key would be `MyLambda.processPayment#${hash}`.

If my understanding is correct, we cannot rely on the resource name only because there might be multiple functions in the runtime code that customers want to make idempotent.

What do you think?